syntax = "proto3";

service MapperService {
    rpc map (MapRequest) returns (MapResponse) {};
}

message MapRequest{
    int32 query = 1;
    string input_location = 2;
    repeated string input_split_files = 3;
    repeated int32 input_split_file_id = 4;
    int32 n_reducers = 5;
}

message MapResponse{
    enum Status {
        FAILED = 0;
        SUCCESS = 1;
    }
    string intermediate_file_location = 1;
    Status status = 2;
    
}